scripts/vim-patch.sh

```diff
@@ -431,25 +439,43 @@ list_missing_vimpatches() {
 
   # Get missing Vim commits
   set +u  # Avoid "unbound variable" with bash < 4.4 below.
-  for vim_commit in $(list_vim_commits "${git_log_args[@]}"); do
+  local vim_commit subject
+  while IFS=' ' read -r line; do
     # Check for vim-patch:<commit_hash> (usually runtime updates).
-    token="vim-patch:${vim_commit:0:7}"
+    token="vim-patch:${line:0:7}"
     if [[ "${tokens[$token]-}" ]]; then
       continue
     fi
 
+    if [[ "$extended_format" == 1 ]]; then
+      vim_commit=${line%% *}
+      subject=${line#* }
+      # Remove "patch 8.0.0902: " prefixes.
+      subject="${subject#patch*: }"
+    else
+      vim_commit=$line
+    fi
+
     vim_tag="${vim_commit_tags[$vim_commit]-}"
     if [[ -n "$vim_tag" ]]; then
       # Check for vim-patch:<tag> (not commit hash).
       patch_number="vim-patch:${vim_tag:1}" # "v7.4.0001" => "7.4.0001"
       if [[ "${tokens[$patch_number]-}" ]]; then
         continue
       fi
-      echo "$vim_tag"
+      if [[ "$extended_format" == 1 ]]; then
+        printf '%s: %s\n' "$vim_tag" "$subject"
+      else
+        printf '%s\n' "$vim_tag"
+      fi
     else
-      echo "$vim_commit"
+      if [[ "$extended_format" == 1 ]]; then
+        printf '%s: %s\n' "$vim_commit" "$subject"
+      else
+        printf '%s\n' "$vim_commit"
+      fi
     fi
-  done
+  done < <(list_vim_commits "${git_log_args[@]}")
```

Comment:
I do not like this to be mangled into the same loop, but it is not too bad either I guess.
Ideas?